browser/brave_shields/brave_shields_web_contents_observer.cc

```diff
@@ -59,21 +59,27 @@ BraveShieldsWebContentsObserver* g_receiver_impl_for_testing = nullptr;
 // npm run test -- brave_browser_tests
 // --filter=BraveContentSettingsAgentImplBrowserTest.*
 void UpdateContentSettingsToRendererFrames(content::WebContents* web_contents) {
-  for (content::RenderFrameHost* frame : web_contents->GetAllFrames()) {
-    Profile* profile =
-        Profile::FromBrowserContext(web_contents->GetBrowserContext());
-    const HostContentSettingsMap* map =
-        HostContentSettingsMapFactory::GetForProfile(profile);
-    RendererContentSettingRules rules;
-    GetRendererContentSettingRules(map, &rules);
-    IPC::ChannelProxy* channel = frame->GetProcess()->GetChannel();
-    // channel might be NULL in tests.
-    if (channel) {
-      mojo::AssociatedRemote<chrome::mojom::RendererConfiguration> rc_interface;
-      channel->GetRemoteAssociatedInterface(&rc_interface);
-      rc_interface->SetContentSettingRules(rules);
-    }
-  }
+  web_contents->GetMainFrame()->ForEachRenderFrameHost(
+      base::BindRepeating([](content::RenderFrameHost* frame) {
+        content::WebContents* web_contents =
+            WebContents::FromRenderFrameHost(frame);
+        DCHECK(web_contents);
+
+        Profile* profile =
+            Profile::FromBrowserContext(web_contents->GetBrowserContext());
```

Comment:
`frame->GetBrowserContext()`, no need for webcontents

Reply:
Fixed in https://github.com/brave/brave-core/commit/2827adb83f4bd7ad66a9f10b74dd85cdfa3a2f38